services/snmptrap/service.go

```diff
@@ -0,0 +1,132 @@
+package snmptrap
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/influxdata/kapacitor"
+	"github.com/k-sone/snmpgo"
+)
+
+type Service struct {
+	targetIp         string
+	targetPort       int
+	community        string
+	version          string
+	stateChangesOnly bool
+	logger           *log.Logger
+}
+
+func NewService(c Config, l *log.Logger) *Service {
+	return &Service{
+		targetIp:         c.TargetIp,
+		targetPort:       c.TargetPort,
+		community:        c.Community,
+		version:          c.Version,
+		stateChangesOnly: c.StateChangesOnly,
+		logger:           l,
+	}
+}
+
+func (s *Service) Open() error {
+	return nil
+}
+
+func (s *Service) Close() error {
+	return nil
+}
+
+func (s *Service) StateChangesOnly() bool {
+	return s.stateChangesOnly
+}
+
+func (s *Service) Alert(trapOid string, dataList [][]interface{}, level kapacitor.AlertLevel) error {
+	// SNMP target address
+	address := s.targetIp + ":" + strconv.Itoa(s.targetPort)
+	// SNMP version
+	var version snmpgo.SNMPVersion
+	switch s.version {
+	case "1":
+		return errors.New("Version 1 not supported yet")
+	case "2c":
+		version = snmpgo.V2c
+	case "3":
+		return errors.New("Version 3 not supported yet")
+	default:
+		return errors.New("Bad snmp version should be: '1', '2c' or '3'")
+	}
```

Comment:
It doesn't add value to add a `version` config option and then only support a single version. How hard is it to support the other versions? From what I saw it looked like the snmp package has support for various versions.